scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/PredictBase.scala

```diff
@@ -0,0 +1,200 @@
+
+  private val logger = LoggerFactory.getLogger(classOf[Predictor])
+
+  protected var batchIndex = inputDescriptors(0).layout.indexOf('N')
+  protected var batchSize = if (batchIndex != -1 ) inputDescriptors(0).shape(batchIndex) else 1
+
+  protected var iDescriptors = inputDescriptors
+
+  inputDescriptors.foreach((f: DataDesc) => require(f.layout.indexOf('N') == batchIndex,
+    "batch size should be in the same index for all inputs"))
+
+
+  if (batchIndex != -1) {
+    inputDescriptors.foreach((f: DataDesc) => require(f.shape(batchIndex) == batchSize,
+      "batch size should be same for all inputs"))
+  } else {
+    // TODO: this is assuming that the input needs a batch
+    iDescriptors = inputDescriptors.map((f : DataDesc) => new DataDesc(f.name,
+    Shape(1 +: f.shape.toVector), f.dtype, 'N' +: f.layout) )
+    batchIndex = 1
+  }
+
+  protected val mxNetHandler = MXNetHandler()
+
+  protected val mod = loadModule()
+
+  /**
+    * This method will take input as IndexedSeq one dimensional arrays and creates
+    * NDArray needed for inference. The array will be reshaped based on the input descriptors.
+    *
+    * @param input : A IndexedSequence of Java one-dimensional array, An IndexedSequence is
+    *              is needed when the model has more than one input/output
+    * @return IndexedSequence array of outputs.
+    */
+  override def predict(input: IndexedSeq[Array[Float]]): IndexedSeq[Array[Float]] = {
+
+    require(input.length == inputDescriptors.length, "number of inputs provided: %d" +
+      " do not match number of inputs in inputDescriptors: %d".format(input.length,
+        inputDescriptors.length))
+
+    for((i, d) <- input.zip(inputDescriptors)) {
+      require (i.length == d.shape.product/batchSize, "number of elements:" +
+        " %d in the input does not match the shape:%s".format( i.length, d.shape.toString()))
+    }
+
+    var inputND: ListBuffer[NDArray] = ListBuffer.empty[NDArray]
+
+    for((i, d) <- input.zip(inputDescriptors)) {
+      val shape = d.shape.toVector.patch(from = batchIndex, patch = Vector(1), replaced = 1)
+
+      inputND += mxNetHandler.execute(NDArray.array(i, Shape(shape)))
+
+    }
+
+    // rebind with batchsize 1
+    if (batchSize != 1) {
+      val desc = iDescriptors.map((f : DataDesc) => new DataDesc(f.name,
+        Shape(f.shape.toVector.patch(batchIndex, Vector(1), 1)), f.dtype, f.layout) )
+      mxNetHandler.execute(mod.bind(desc, outputDescriptors, forceRebind = true,
+        forTraining = false))
+    }
+
+    val resultND = mxNetHandler.execute(mod.predict(new NDArrayIter(inputND.toIndexedSeq)))
+
+    val result = resultND.map((f : NDArray) => f.toArray)
+
+    mxNetHandler.execute(inputND.foreach((f: NDArray) => f.dispose))
+    mxNetHandler.execute(resultND.foreach((f: NDArray) => f.dispose))
+
+    // rebind to batchSize
+    if (batchSize != 1) {
+      mxNetHandler.execute(mod.bind(inputDescriptors, forTraining = false, forceRebind = true))
+    }
+
```

Comment:
extra empty line

Reply:
Is it standard to not have blank lines, I tend to put blank lines so that it is more readable.